### PR TITLE
Fat-186 creating lars index

### PR DESCRIPTION
**Pull request by mapaux**

### CODE REVIEW

File: src/Sfa.Das.Sas.Indexer.ApplicationServices/Lars/Services/LarsIndexer.cs [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+            var larsData = _metaDataHelper.GetAllApprenticeshipLarsMetaData();
+
+            //TODO: index LARS data
+            await IndexStandards(indexName, larsData.Standards).ConfigureAwait(false);
+            await IndexFrameworks(indexName, larsData.Frameworks).ConfigureAwait(false);
+            await IndexFundingMetadata(indexName, larsData.FundingMetaData).ConfigureAwait(false);
+            await IndexFrameworkAimMetaData(indexName, larsData.FrameworkAimMetaData).ConfigureAwait(false);
+            await IndexLearningDeliveryMetaData(indexName, larsData.LearningDeliveryMetaData).ConfigureAwait(false);
+            await IndexApprenticeshipComponentTypeMetaData(indexName, larsData.ApprenticeshipComponentTypeMetaData).ConfigureAwait(false);
+            await IndexApprenticeshipFunding(indexName, larsData.ApprenticeshipFunding).ConfigureAwait(false);
```
Comment (sachpatel) [inline src/Sfa.Das.Sas.Indexer.ApplicationServices/Lars/Services/LarsIndexer.cs]: Task.WaitAll?

File: src/Sfa.Das.Sas.Indexer.ApplicationServices/Lars/Services/LarsIndexer.cs [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+            {
+                _log.Error(ex, "Error indexing LARS LearningDeliveryMetaData");
+            }
+        }
+
+        private async Task IndexApprenticeshipFunding(string indexName, IEnumerable<ApprenticeshipFunding> larsDataApprenticeshipFunding)
+        {
+            try
+            {
+                _log.Debug("Indexing " + larsDataApprenticeshipFunding.Count() + " learning delivery metadata details");
```
Comment (sachpatel) [inline src/Sfa.Das.Sas.Indexer.ApplicationServices/Lars/Services/LarsIndexer.cs]: Should be Apprenticeship Funding details.

File: src/Sfa.Das.Sas.Indexer.ApplicationServices/Shared/IndexerService.cs [hunk 3]
```diff
@@ -76,5 +78,15 @@ private void PauseWhileIndexingIsBeingRun()
             var time = _indexSettings.PauseTime;
             Thread.Sleep(int.Parse(time));
         }
+
+        private string GetIndexTypeName(Type type)
+        {
+            if (type == typeof(IMaintainProviderIndex))
```
Comment (sachpatel) [inline src/Sfa.Das.Sas.Indexer.ApplicationServices/Shared/IndexerService.cs]: Could put a switch statement.

File: src/Sfa.Das.Sas.Infrastructure/Shared/Elasticsearch/ElasticsearchMapper.cs [hunk 4]
```diff
@@ -52,6 +55,21 @@ public StandardDocument CreateStandardDocument(StandardMetaData standard)
             return doc;
         }
 
+        public StandardLars CreateLarsStandardDocument(LarsStandard standard)
+        {
+            var doc = new StandardLars
```
Comment (scottcowan) [inline src/Sfa.Das.Sas.Infrastructure/Shared/Elasticsearch/ElasticsearchMapper.cs]: you can just return this

File: src/Sfa.Das.Sas.Infrastructure/Shared/Elasticsearch/ElasticsearchMapper.cs [hunk 5]
```diff
@@ -86,6 +104,111 @@ public FrameworkDocument CreateFrameworkDocument(FrameworkMetaData frameworkMeta
             return doc;
         }
 
+        public FrameworkLars CreateLarsFrameworkDocument(FrameworkMetaData frameworkMetaData)
+        {
+            // Trim off any whitespaces in the title or the Pathway Name
+            frameworkMetaData.NasTitle = frameworkMetaData.NasTitle?.Trim();
+            frameworkMetaData.PathwayName = frameworkMetaData.PathwayName?.Trim();
+
+            var doc = new FrameworkLars
```
Comment (scottcowan) [inline src/Sfa.Das.Sas.Infrastructure/Shared/Elasticsearch/ElasticsearchMapper.cs]: just return

File: src/Sfa.Das.Sas.Tools.MetaDataCreationTool/Services/LarsDataService.cs [hunk 6]
```diff
@@ -40,6 +42,195 @@ public sealed class LarsDataService : ILarsDataService
...
+
+            var learningDeliveryMetadata = GetLarsLearningDeliveryMetaData(zipStream);
+
+            var fundingMetadata = GetLarsFundingMetaData(zipStream);
+
+            var apprenticeshipFunding = GetApprenticeshipFunding(zipStream);
+
+            CloseStream(zipStream);
+
+            var larsData = new LarsData
```
Comment (scottcowan) [inline src/Sfa.Das.Sas.Tools.MetaDataCreationTool/Services/LarsDataService.cs]: just return

**Pull request merged**